### PR TITLE
add atom codecov

### PR DESCRIPTION
**Pull request by purelind**
* add atom-ut-codecov for same coverage against tidb_ghpr_coverage

### CODE REVIEW

File: atom-jobs/atom-ut-codecov.groovy [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                                sh """
+                                curl -LO ${FILE_SERVER_URL}/download/cicd/ci-tools/codecov
+                                chmod +x codecov
+                                ./codecov -f 'coverage_report/*.coverage'  -t ${CODECOV_TOKEN} -C ${COMMIT_ID} -P ${PULL_REQUEST_ID} -b ${BUILD_NUMBER}
+                                """
+                            } else {
+                                sh """
+                                curl -LO ${FILE_SERVER_URL}/download/cicd/ci-tools/codecov
+                                chmod +x codecov
+                                ./codecov -f 'coverage_report/*.coverage' -t ${CODECOV_TOKEN} -C ${COMMIT_ID} -b ${BUILD_NUMBER} -B ${BRANCH}
```
Comment (dianqihanwangzi) [inline atom-jobs/atom-ut-codecov.groovy:156]: if so, every task should generate cov file like this. We can use params passed in.

File: atom-jobs/atom-ut-codecov.groovy [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                                """
+                            }
+                        }
+                        sleep(time:120,unit:"SECONDS")
+                        def response = httpRequest Authorization: CODECOV_API_TOKEN, url: "https://codecov.io/api/gh/pingcap/tidb/commit/${COMMIT_ID}"
+                        println('Status: '+response.status)
+                        def obj = readJSON text:response.content
+                        if (response.status == 200) {
+                            println(obj.commit.totals)
+                            lines_coverage_rate = "Lines coverage ${obj.commit.totals.c.toFloat().round(2)}%."
```
Comment (dianqihanwangzi) [inline atom-jobs/atom-ut-codecov.groovy:166]: Should we fail build when not reach coverage rate？

Comment (purelind) [reply atom-jobs/atom-ut-codecov.groovy:166]: if ut test failed, current coverage is not correct, so we not display coverage when ut failed

File: atom-jobs/atom-ut-codecov.groovy (hunk 1 above)
Comment (purelind) [reply atom-jobs/atom-ut-codecov.groovy:156]: emm. some debug code here. i will modify here

File: atom-jobs/atom-ut-codecov.groovy [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            for (varString in varStrings) {
+                keyAndValue = varString.split(":")
+                if (keyAndValue.length == 2) {
+                    credentialList.push(string(credentialsId: keyAndValue[0], variable: keyAndValue[1]))
+                }
+            }
+
+            stage("Test") {
+                dir("${ws}/${REPO}") {
+                    sh TEST_CMD
```
Comment (dianqihanwangzi) [inline atom-jobs/atom-ut-codecov.groovy:135]: Should also pass credentialList to test stage?

**Review by dianqihanwangzi — APPROVED** [review]
lgtm

**Pull request merged**